spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -189,9 +192,8 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
 
     boolean cboEnabled =
         Boolean.parseBoolean(spark.conf().get(SQLConf.CBO_ENABLED().key(), "false"));
-    Map<NamedReference, ColumnStatistics> colStatsMap = null;
+    Map<NamedReference, ColumnStatistics> colStatsMap = Maps.newHashMap();
```

Comment:
In Spark side, it has a `columnStats().isEmpty`, so I changed back to `Maps.newHashMap()` to avoid the NPE.

Reply:
What I meant here was to have it be Collection EmptyMap unless cbo is enabled, probably a premature optimization. So feel free to ignore

Ie:
```java
Map colStatsMap;
  if (readConf && cbo)
     colStatsMap = Maps.newHashMap()
  else 
    colStatsMap = Collections.emptyMap()
```

Reply:
Got it. Changed!